torch/utils/flop_counter.py

```diff
@@ -0,0 +1,306 @@
+import torch
+from torch.utils._pytree import tree_map
+from typing import List, Any, Dict
+from collections import defaultdict
+from torch.utils._python_dispatch import TorchDispatchMode
+import functools
+
+aten = torch.ops.aten
+
+def get_shape(i):
+    if isinstance(i, torch.Tensor):
+        return i.shape
+    return i
+
+def prod(x):
+    res = 1
+    for i in x:
+        res *= i
+    return res
+
+def mm_flop(input_shapes: List[Any], _=None) -> int:
+    """
+    Count flops for matmul.
+    """
+    # Inputs should be a list of length 2.
+    # Inputs contains the shapes of two matrices.
+    assert len(input_shapes) == 2, input_shapes
+    a_shape, b_shape = input_shapes
+    m, k = a_shape
+    k2, n = b_shape
+    assert k == k2
+    # NB(chilli): Should be 2 * k - 1 technically for FLOPs.
+    return m * n * 2 * k
+
+def addmm_flop(input_shapes: List[Any], _=None) -> int:
+    """
+    Count flops for addmm
+    """
+    assert len(input_shapes) == 3
+    _, a_shape, b_shape = input_shapes
+    mm_flops = mm_flop([a_shape, b_shape])
+    batch_size, _ = a_shape
+    _, output_dim = b_shape
+    return mm_flops + batch_size * output_dim
```

Comment:
wait, why? `mm` flops already account for this?

Reply:
I was just adding the addition flops, although maybe this is unneeded 🤔

Reply:
Decided to remove them for consistency.